app/routes/circulars/circulars.lib.ts

```diff
@@ -76,9 +80,17 @@ const subjectMatchers: SubjectMatcher[] = [
     /Baksan\sNeutrino\sObservatory\sAlert[.\s_-]*(\d{6}.\d{2})/i,
     ([, id]) => `Baksan Neutrino Observatory Alert ${id}`,
   ],
-  [/EP[.\s_-]*(\d{6}[a-z])/i, ([, id]) => `EP${id}`],
-  [/FRB[.\s_-]*(\d{8}[a-z])/i, ([, id]) => `FRB ${id}`.toUpperCase()],
+  [/EP[.\s_-]*(\d{6}[a-z])/i, ([, id]) => `EP${id.toLowerCase()}`],
+  [
+    /SN[.\s_-]*(\d{4}[a-zA-Z]*)/i,
+    ([, id]) => `SN ${id.length == 5 ? id.toUpperCase() : id.toLowerCase()}`,
+  ],
+  [/GW[.\s_-]*(\d{6})/i, ([, id]) => `GW${id}`],
+  [/FRB[.\s_-]*(\d{8}[a-zA-Z])/i, ([, id]) => `FRB ${id}`.toUpperCase()],
```

Comment:
```suggestion
  [/FRB[.\s_-]*(\d{8}[a-z])/i, ([, id]) => `FRB ${id}`.toUpperCase()],
```

... and similarly for all of the other new regular expressions, because you are using the `/i` (case insensitive) flag.

Reply:
That would be an inadvisable change. It would not normalize optional or wildcard matches appropriately.

I know it's counter intuitive and that it theoretically *should* match since it's case insensitive, but some regex engines are not able to determine with the case insensitive flag if matching on the lower case or upper case is a better match and since they are optional or wildcard, it just takes the digits and leaves off the letter(s) since that is also a valid match. It's just one of those regex gotchas that you have to be aware of.
<img width="568" alt="Screenshot 2025-04-17 at 9 55 03 AM" src="https://github.com/user-attachments/assets/4acd0db0-46bd-484f-82ee-7d5ff5398176" />
<img width="607" alt="Screenshot 2025-04-17 at 9 55 16 AM" src="https://github.com/user-attachments/assets/964865ab-b6d3-4776-8223-5b17305d643b" />
<img width="642" alt="Screenshot 2025-04-17 at 9 55 25 AM" src="https://github.com/user-attachments/assets/e86f3ce2-6f92-4dd8-a232-789e0091b2d8" />
<img width="582" alt="Screenshot 2025-04-17 at 9 55 37 AM" src="https://github.com/user-attachments/assets/efd12c9a-383b-443f-ac1f-1679a872d261" />
<img width="586" alt="Screenshot 2025-04-17 at 9 55 44 AM" src="https://github.com/user-attachments/assets/c5dc4c17-720c-4112-91f9-dacc7239b598" />

Reply:
> I know it's counter intuitive and that it theoretically _should_ match since it's case insensitive, but some regex engines are not able to determine with the case insensitive flag if matching on the lower case or upper case is a better match and since they are optional or wildcard, it just takes the digits and leaves off the letter(s) since that is also a valid match. It's just one of those regex gotchas that you have to be aware of.

I am not aware of which regular expressions behave that way, but all that matters for our application is how the ECMAScript-specified regular expression processing (and more specifically, Node.js's regular expression processing) treats case-insensitive matches. Node.js does respect the `//i` flag, as documented:

```
$ node
Welcome to Node.js v22.14.0.
Type ".help" for more information.
> /AT[.\s_-]*(\d{4}[a-z]*)/i.exec('AT 2023LCRSJQ: VLA radio detection')
[
  'AT 2023LCRSJQ',
  '2023LCRSJQ',
  index: 0,
  input: 'AT 2023LCRSJQ: VLA radio detection',
  groups: undefined
]
```

The failures in those screenshots above don't contradict this. There is a bug in `parseEventFromSubject`: we are dynamically constructing new regular expressions from the original sources, and we failed to copy the original regular expressions' flags. This is fixed by https://github.com/nasa-gcn/gcn.nasa.gov/pull/3049.